code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -53,14 +53,15 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	var/list/guardian_overlays[GUARDIAN_TOTAL_LAYERS]
 	var/reset = 0 //if the summoner has reset the guardian already
 	var/cooldown = 0
-	var/mob/living/summoner
+	/// weakref to our summon which SHOULD be a living mob.
+	var/datum/weakref/weak_summoner
```

Comment:
whats wrong with the name summoner? more clear, even if it is a weakref

Reply:
so summoner can be used for the resolved weak summoner

Reply:
weakrefs generally just have ``_ref`` at the end